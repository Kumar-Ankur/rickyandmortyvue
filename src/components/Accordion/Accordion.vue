<template>
  <section id="app" class="section container">
    <article class="message" :class="accordionClasses">
      <div class="message-header" @click="toggleAccordion">{{ text }}</div>
      <div class="message-body">
        <div class="message-content">
          <app-species-view></app-species-view>
            <app-gender-view></app-gender-view>
            <app-origin-view></app-origin-view>
        </div>
      </div>
    </article>
  </section>
</template>

<script>
import SpeciesView from '../SpeciesView/SpeciesView.vue';
import GenderView from '../GenderView/GenderView.vue';
import OriginView from '../OriginView/OriginView.vue';

export default {
    props: ['text'],
  data() {
    return {
      isOpen: false
    };
  },
  methods: {
    toggleAccordion() {
      this.isOpen = !this.isOpen
    }
  },
  computed: {
    accordionClasses: function() {
      return {
        "is-closed": !this.isOpen,
        "is-primary": this.isOpen,
        "is-dark": !this.isOpen
      };
    }
  },
  components: {
    appSpeciesView: SpeciesView,
    appGenderView: GenderView,
    appOriginView: OriginView
  }
};
</script>

<style scoped>
.message {
  width: 90vw;
  margin-top: 3rem;
  background: linear-gradient(to right, #3c3b3f, #605c3c);
  color: white;
  cursor: pointer;
  padding: 18px;
  align-items: center;
  transition: ease-in-out 3s all;
}
.message-header {
  cursor: pointer;
  font-size: 1.5rem;
}
.message-body {
  padding: 0;
  max-height: 50em;
  overflow: hidden;
  transition: 0.3s ease all;
}
.is-closed .message-body {
  max-height: 0;
}
.message-content {
  padding: 20px;
}
</style>