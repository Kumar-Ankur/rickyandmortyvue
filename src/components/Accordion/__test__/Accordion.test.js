import { shallowMount } from "@vue/test-utils";
import Accordion from '../Accordion.vue';

describe('unit test case for accordion component', () => {
    test('generate snapshot test', () => {
        const wrapper = shallowMount(Accordion, {
            propsData: {
                text: 'Test'
            }
        })
        expect(wrapper.html()).toMatchSnapshot();
    })
})