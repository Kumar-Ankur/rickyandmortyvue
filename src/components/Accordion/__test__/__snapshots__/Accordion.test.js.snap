// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test case for accordion component generate snapshot test 1`] = `
"<section id=\\"app\\" class=\\"section container\\">
  <article class=\\"message is-closed is-dark\\">
    <div class=\\"message-header\\">Test</div>
    <div class=\\"message-body\\">
      <div class=\\"message-content\\">
        <app-gender-view-stub></app-gender-view-stub>
        <app-origin-view-stub></app-origin-view-stub>
        <app-species-view-stub></app-species-view-stub>
      </div>
    </div>
  </article>
</section>"
`;
